plat/kvm/Config.uk

```diff
@@ -1,20 +1,24 @@
 menuconfig PLAT_KVM
-       bool "KVM guest"
-       default n
-       depends on (ARCH_X86_64 || (ARCH_ARM_64 && !HAVE_SYSCALL))
-       select LIBUKDEBUG
-       select LIBUKALLOC
-       select LIBUKTIMECONV
-       select LIBNOLIBC if !HAVE_LIBC
-       select LIBFDT if ARCH_ARM_64
-       select ELF64_TO_32 if ARCH_X86_64
-       select LIBUKRELOC if OPTIMIZE_PIE
-       select LIBUKOFW if ARCH_ARM_64
-       imply LIBUKBUS_PCI if LIBUKBUS
-       imply LIBUKBUS_PLATFORM if (LIBUKBUS && ARCH_ARM_64)
-       select LIBUKINTCTLR if ARCH_X86_64
-       help
-                Create a Unikraft image that runs as a KVM guest
+	bool "KVM guest"
+	default n
+	depends on (ARCH_X86_64 || (ARCH_ARM_64 && !HAVE_SYSCALL))
+	select LIBUKDEBUG
+	select LIBUKALLOC
+	select LIBUKTIMECONV
+	select LIBNOLIBC if !HAVE_LIBC
+	select LIBFDT if ARCH_ARM_64
+	select ELF64_TO_32 if ARCH_X86_64
+	select LIBUKRELOC if OPTIMIZE_PIE
+	select LIBUKOFW if ARCH_ARM_64
+	imply  LIBUKBUS_PCI if LIBUKBUS
```

Comment:
I spent big part of yesterday discussing with @skuenzer about Kconfig. It all started by the fact that nothing selects `libukbus` right now. There are some additional requirements, eg not to show virtio-pci or virtio-9p to the menu if they are not supported by a given VMM (eg Firecracker), not to show the virtio-bus and virtio transports when no device is selected, show only the relevant transports that are supported by a given VMM and so on.

Based on these:
1. Please add a commit to this PR to add `HAVE_PCI`. We need this so that we can hide PCI by platforms that don't have this ability.
```
diff --git a/drivers/ukbus/pci/Config.uk b/drivers/ukbus/pci/Config.uk
index 57387898..77b5f767 100644
--- a/drivers/ukbus/pci/Config.uk
+++ b/drivers/ukbus/pci/Config.uk
@@ -1,7 +1,11 @@
+config HAVE_PCI
+       bool
+
 config LIBUKBUS_PCI
        bool "PCI bus"
+       depends on HAVE_PCI
        depends on (ARCH_X86_64 || ARCH_ARM_64)
-       depends on LIBUKBUS
+       select LIBUKBUS
        help
                PCI bus driver for probing and operating PCI devices
 ```
 You will see below how that is used.

2. Add a commit in this series to have libofw to `select` its dependencies. The idea is to allows chained dependency resolution and use `depends on` only when there is a dependency loop, as we discussed before.
```
diff --git a/lib/ukofw/Config.uk b/lib/ukofw/Config.uk
index 8e2d1816..0e05ca25 100644
--- a/lib/ukofw/Config.uk
+++ b/lib/ukofw/Config.uk
@@ -1,4 +1,4 @@
 config LIBUKOFW
        bool "ukofw: Device tree helper functions"
        default n
-       depends on LIBFDT
+       select LIBFDT
```

3. In `virtio`. Add a top-level `Config.uk` as:
```
config VIRTIO_DEVICE
	bool
	default y if (LIBVIRTIO_9P || LIBVIRTIO_BLK || LIBVIRTIO_NET)
```
This will be used as syntactic sugar below to avoid typing all devices one by one. Yes yes I have asked you a few times to change things around, I'm sorry, things are still evolving. This is the last time I promise.

4. In `virtio-pci`. Have it select it's dependenies (ie `LIBUKBUS_PCI`) and make it available only to devices that support PCI (`HAVE_PCI`). Finally we also have the transports (`virtio-pci`, `virtio-mmio`) to no longer select `virtio-bus`, but instead make them invisible until `virtio-bus` is selected. You will see below under which conditions that happens.
```
diff --git a/drivers/virtio/pci/Config.uk b/drivers/virtio/pci/Config.uk
index 7f63e1ad..4afcc642 100644
--- a/drivers/virtio/pci/Config.uk
+++ b/drivers/virtio/pci/Config.uk
@@ -1,6 +1,7 @@
 config LIBVIRTIO_PCI
        bool "Virtio PCI transport"
-       depends on LIBUKBUS_PCI
-       select LIBVIRTIO_BUS
+       depends on VIRTIO_DEVICE
+       depends on HAVE_PCI
+       select LIBUKBUS_PCI
        help
                Support virtio devices on PCI bus
```
5. In `virtio-mmio`, similar changes like `virito-pci`, but since it doesn't have a parent we add `HAVE_MMIO` here:
```
diff --git a/drivers/virtio/mmio/Config.uk b/drivers/virtio/mmio/Config.uk
index 4bfce30c..85e66c7f 100644
--- a/drivers/virtio/mmio/Config.uk
+++ b/drivers/virtio/mmio/Config.uk
@@ -1,9 +1,11 @@
+config HAVE_MMIO
+       bool
+
 config LIBVIRTIO_MMIO
-    bool "Virtio MMIO transport"
-    depends on LIBUKBUS
-    depends on ARCH_ARM_64
-    select LIBVIRTIO_BUS
-    select LIBUKOFW
-    select LIBUKINTCTLR_GIC
-    help
-        Virtio MMIO transport.
+       bool "Virtio MMIO transport"
+       depends on ARCH_ARM_64
+       depends on VIRTIO_DEVICE
+       select LIBUKOFW
+       select LIBUKINTCTLR_GIC
+       help
+               Virtio MMIO transport.
```
6. In `virtio-bus` make it visible if at least one virtio device is enabled (`HAVE_VIRTIO`), have it `select` its dependency of `LIBUKBUS`. Here we don't need to use `depends on` as there is does no circular dependency, and in fact solves the problem that nothing selected `ukbus` until now.  Also enable by defaul tthe transports (mmio / pci) for those platforms that can use them (`HAVE_[MMIO,PCI]`).
```
diff --git a/drivers/virtio/bus/Config.uk b/drivers/virtio/bus/Config.uk
index d2f594d2..dc107a26 100644
--- a/drivers/virtio/bus/Config.uk
+++ b/drivers/virtio/bus/Config.uk
@@ -1,6 +1,9 @@
 config LIBVIRTIO_BUS
        bool "Virtio ukbus driver"
-       depends on LIBUKBUS
+       depends on VIRTIO_DEVICE
        select LIBUKALLOC
+       select LIBUKBUS
        select LIBUKSGLIST
        select LIBVIRTIO_RING
+       imply LIBVIRTIO_PCI if HAVE_PCI
+       imply LIBVIRTIO_MMIO if HAVE_MMIO
```

7. Now on to the devices. These don't need any changes, as they `select` their direct dependencies, and otherwise mind their own business. No dependency loops here either as they already `depend on` their API library (`ukblkdev`, `uknetdev`, `libuk9p`).

8. Finally the on the platform declare which features are supported by each VMM, and have the drivers selected if their corresponding feature is requested by the application.
```
diff --git a/plat/kvm/Config.uk b/plat/kvm/Config.uk
index aca9bef7..a14d5332 100644
--- a/plat/kvm/Config.uk
+++ b/plat/kvm/Config.uk
@@ -10,13 +10,11 @@ menuconfig PLAT_KVM
        select ELF64_TO_32 if ARCH_X86_64
        select LIBUKRELOC if OPTIMIZE_PIE
        select LIBUKOFW if ARCH_ARM_64
-       imply  LIBUKBUS_PCI if LIBUKBUS
-       imply  LIBVIRTIO_PCI if LIBUKBUS
-       imply  LIBVIRTIO_MMIO if (LIBUKBUS && ARCH_ARM_64)
-       imply  LIBVIRTIO_BLK if LIBUKBLKDEV
-       imply  LIBVIRTIO_NET if LIBUKNETDEV
-       imply  LIBUKBUS_PLATFORM if (LIBUKBUS && ARCH_ARM_64)
        select LIBUKINTCTLR if ARCH_X86_64
+       imply LIBUKBUS_PLATFORM if ARCH_ARM_64
+       imply LIBVIRTIO_9P if LIBUK9P
+       imply LIBVIRTIO_NET if LIBUKNETDEV
+       imply LIBVIRTIO_BLK if LIBUKBLKDEV
        help
                Create a Unikraft image that runs as a KVM guest
 
@@ -96,11 +94,13 @@ choice
 config KVM_VMM_QEMU
        bool "QEMU"
        imply LIBUKINTCTLR_GICV2
-       imply LIBVIRTIO_9P if LIBUK9P
+       select HAVE_PCI
+       select HAVE_MMIO
 
 config KVM_VMM_FIRECRACKER
        bool "Firecracker"
        select KVM_BOOT_PROTO_LXBOOT
+       select HAVE_MMIO
        imply LIBUKINTCTLR_GICV3
        depends on ARCH_X86_64 || ARCH_ARM_64
        
```